src/Atmos/Parameterizations/SubgridScaleTurbulence/SubgridScaleTurbulence.jl

```diff
@@ -0,0 +1,275 @@
+    # For a sphere: the arc length provides one approximation of many
+    Δ = cbrt(Δ1 * Δ2 * Δ3)
+    Δ_sorted = sort([Δ1, Δ2, Δ3])  
+    # Get smallest two dimensions
+    Δ_s1 = Δ_sorted[1]
+    Δ_s2 = Δ_sorted[2]
+    a1 = Δ_s1 / max(Δ1,Δ2,Δ3) 
+    a2 = Δ_s2 / max(Δ1,Δ2,Δ3) 
+    # In 3D we compute a scaling factor for anisotropic grids
+    f_anisotropic = 1 + 2/27 * ((log(a1))^2 - log(a1)*log(a2) + (log(a2))^2)
+    Δ = Δ*f_anisotropic
+    Δsqr = Δ * Δ
+    return Δsqr
+  end
+  
+  """
+    anisotropic_lengthscale_2D(Δ1, Δ2)
+    return Δ², the equivalent anisotropic lengthscale squared
+
+  For a 2-D problem, compute the anisotropic length-scale
+  given the two local grid dimensions. For example, edge length
+  for cube meshes and arc length for curved grids. The local element
+  length can be computed within the auxiliary state initialisation kernel
+  and passed to the auxiliary state in a manner similar to the coordinate
+  terms.
+  """
+
+  function anisotropic_lengthscale_2D(Δ1, Δ2)
+    # Order of arguments does not matter.
+    Δ = min(Δ1, Δ2)
+    Δsqr = Δ * Δ
+    return Δsqr
+  end
+  
+  """
+    geo_mean_lengthscale_3D(Δ1, Δ2, Δ3)
+    return Δ², the geometric mean lengthscale squared
+
+  For a 3-D problem, compute the standard, geometric mean
+  lengthscale based on the three coordinate dimensions 
+  Δ1, Δ2, Δ3. The local element length can be computed 
+  within the auxiliary state initialisation kernel
+  by passing in the metric terms as arguments and 
+  stored therein for use throughout the driver.
+  """
+  function standard_lengthscale_3D(Δ1,Δ2,Δ3)
+    Δ = cbrt(Δ1 * Δ2 * Δ3) 
+    Δsqr = Δ * Δ
+    return Δsqr
+  end
+  
+  """
+    geo_mean_lengthscale_2D(Δ1, Δ2)
+    return Δ², the geometric mean lengthscale squared
+
+  For a 2-D problem, compute the standard, geometric mean
+  lengthscale based on the three coordinate dimensions 
+  Δ1, Δ2, Δ3. The local element length can be computed 
+  within the auxiliary state initialisation kernel
+  by passing in the metric terms as arguments and 
+  stored therein for use throughout the driver.
+  """
+  function geo_mean_lengthscale_2D(Δ1, Δ2)
+    Δ = sqrt(Δ1 * Δ2)
+    Δsqr = Δ * Δ
+    return Δsqr
+  end
+
+  """
+    strainrate_tensor_components(dudx, dudy, dudz, dvdx, dvdy, dvdz [, dwdx, dwdy, dwdz])
+    return (S11, S22, S33, S12, S13, S23, SijSij)
+    Sij are components of the strain-rate tensor, and SijSij is the tensor-inner-product
+
+  Compute components of strain-rate tensor from velocity gradient terms provided in 
+  driver. Note that the gradient terms are computed in Cartesian space even on the spherical
+  grid, with velocities and other key variables projected onto spherical coordinates for 
+  visualisation 
+
+  Dij = ∇u .................................................. [1]
+  Sij = 1/2 (∇u + (∇u)ᵀ) .....................................[2]
+  τij = 2 * ν_e * Sij ........................................[3]
```

Comment:
Can we use the sign convention here that gives us momentum fluxes, i.e., `\tau_ij = -2 \nu_e S_ij` (as in design document)?